### PR TITLE
[JENKINS-51820] Removing Java Web Start support

### PR DESCRIPTION
**Pull request by jglick**
See [JENKINS-51820](https://issues.jenkins.io/browse/JENKINS-51820).

Amends #3766.

Preparation for [dropping Java 8 support](https://github.com/jenkinsci/jep/blob/master/jep/236/README.adoc#library-updates), since we did not support JWS on newer Java versions anyway.

* Changes to the agent JAR as noted in upstream https://github.com/jenkinsci/remoting/pull/532.
* Removing the JWS button from the GUI displayed on Java 8.
* Stripping out parts of `jenkins-agent.jnlp` which were interpreted by the real `javaws` but ignored by the `java -jar agent.jar -jnlpUrl …` parser.
* Removing [agent installer modules](https://github.com/jenkinsci/jep/blob/master/jep/230/README.adoc#marking-slave-installer-as-detached) as they only work in GUI ~ JWS mode, and among other concerns add overhead to every agent connection even when unused. These repositories could then be archived, or optionally redesigned to run as standalone tools (especially the Windows one which is the most likely to be used):
  * https://github.com/jenkinsci/slave-installer-module
  * https://github.com/jenkinsci/windows-slave-installer-module
  * https://github.com/jenkinsci/systemd-slave-installer-module
  * https://github.com/jenkinsci/launchd-slave-installer-module
  * https://github.com/jenkinsci/upstart-slave-installer-module
* Removing https://github.com/jenkinsci/jenkins/blob/db4d8b40c81032db45c5c205d4249b82950cf4ad/core/src/main/resources/hudson/slaves/JNLPLauncher/help-vmargs.html#L2-L4 since the only remaining usage was a prompt for you to run Java yourself: https://github.com/jenkinsci/jenkins/blob/db4d8b40c81032db45c5c205d4249b82950cf4ad/core/src/main/resources/hudson/slaves/JNLPLauncher/main.jelly#L41 Basically the only remaining function of the `vmargs` field would have been as a “note to self” field where you can write down arguments you would want to pass next time you connect, if you are having a hard time remembering them or something. (Previously it actually controlled what JVM args `javaws` would pass to the agent. We cannot do that when you run `java -jar agent.jar` because the JVM is already started.)

Sanity tested on Linux using inbound TCP and WebSocket agents running simple builds, with and without security enabled, with both Java 8 and 11 for both controller and agent.

### Proposed changelog entries

* Remove [Java Web Start](https://en.wikipedia.org/wiki/Java_Web_Start) support for launching inbound agents, along with the GUI mode, the platform-specific agent installers and the JAR signature.

### Proposed upgrade guidelines

Jenkins no longer supports attaching a static inbound agent by clicking the **☕ Launch** button from an agent machine’s web browser when running the controller on Java 8, as Java Web Start has been removed from newer versions of most distributions. You should instead download the agent JAR file from the provided link and run the supplied command (`java -jar agent.jar -jnlpUrl …`) on the agent machine. The **JVM options** field was removed from inbound launcher configuration as it would no longer have any effect other than adjusting the suggested command.

The control window displayed when using a Java Web Start agent is also removed. Its main function was to be closed (now done by simply terminating the shell process). It also offered a menu with platform-specific agent installers that would offer to create system services to make the agent permanent; you can accomplish something similar in many ways without this GUI, according to your operating system.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

**Review by basil — APPROVED** [review]
Very nice! Presumably https://github.com/jenkinsci/netx can be archived once this PR is merged. I also created https://github.com/jenkinsci/jenkins/pull/6545 and https://github.com/jenkinsci/extras-executable-war/pull/52 with some additional JNLP-related cleanup.

**Comment by jeffret-b** [comment]
> Basically the only remaining function of the `vmargs` field is as a “note to self” field where you can write down arguments you would want to pass next time you connect, if you are having a hard time remembering them or something. (Previously it actually controlled what JVM args `javaws` would pass to the agent. We cannot do that when you run `java -jar agent.jar` because the JVM is already started.) Perhaps it would make sense to delete this now.

I would go ahead and delete this now.

**Review by jeffret-b — APPROVED** [review]
I am strongly in favor of this improvement. I tried a couple of times previously to push this removal through, but wasn't able then to overcome the resistance. There really is no reason to continue supporting JWS, especially as we move forward with Java versions.

This PR covers all of the different areas related to JWS that I can recall. It looks like it handles the changes correctly.

**Review by basil — APPROVED** [review]

File: pom.xml
Comment (timja) [inline pom.xml:94]:
```suggestion
    <remoting.version>4.14</remoting.version>
```

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by timja** [comment]
> Hold on: [jenkinsci/remoting#532 (comment)](https://github.com/jenkinsci/remoting/pull/532#issuecomment-1125014644)

resolved ^^

**Pull request merged**
